src/main/java/com/wise23/chariteed/controller/PatientGeneratorController.java

```diff
@@ -0,0 +1,103 @@
+package com.wise23.chariteed.controller;
+
+import com.wise23.chariteed.QRCodeGenerator;
+import com.wise23.chariteed.model.Role;
+import com.wise23.chariteed.model.User;
+import com.wise23.chariteed.model.PatientGenerator;
+import com.wise23.chariteed.service.PatientService;
+import com.wise23.chariteed.service.UserService;
+
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import org.hl7.fhir.r4.model.Patient;
+
+import com.google.gson.*;
+import com.google.zxing.WriterException;
+
+import java.util.Random;
+import java.io.IOException;
+import java.util.List;
+
+@Controller
+public class PatientGeneratorController {
+
+    PatientService patientService = new PatientService();
+
+    @Autowired
+    UserService userService;
+
+    @GetMapping("admin/dashboard/generatePatient")
+    public String generatePatientForm(Model model) {
+        model.addAttribute("generatePatient", new PatientGenerator());
+        return "admin/dashboard/generatePatient";
+    }
+
+    @PostMapping("admin/dashboard/generatePatient")
+    public String generatePatientSubmit(@ModelAttribute PatientGenerator generator, Model model)
+            throws WriterException, IOException {
+        String id = Long.toString(generator.getId());
+        model.addAttribute("generatePatient", generator);
+
+        Patient patient = null;
+
+        // In case that the ID is not existing.
+        try {
+            patient = patientService.client.read().resource(Patient.class).withId(id).execute();
+        } catch (Exception e) {
+            System.out.println("ERROR: Patient ID does not exist.");
+            return "admin/dashboard/error";
+        }
+
+        String patientData = patientService.fhirContext.newJsonParser().setPrettyPrint(true)
+                .encodeResourceToString(patient);
+
+        generator.setPassword(generatePassword(patientData));
+
+        // QR code generation
+        QRCodeGenerator.createQRImage(id);
+
+        JsonObject name = JsonParser.parseString(patientData).getAsJsonObject().get("name").getAsJsonArray().get(0)
+                .getAsJsonObject();
+
+        User user = new User();
```

Comment:
I would move it to User class. 
Also, if u want to use something cool, there is a Builder pattern 
https://www.digitalocean.com/community/tutorials/builder-design-pattern-in-java

Reply:
What is _it_? The Json parsing?

Reply:
@JannesVolkens by it, I meant the process of creation a user object, populating its fields with data.

Reply:
Done